Mathlib/RingTheory/HopfAlgebra.lean

```diff
@@ -0,0 +1,76 @@
+/-
+Copyright (c) 2024 Ali Ramsey. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Ali Ramsey
+-/
+import Mathlib.RingTheory.Bialgebra
+
+/-!
+# Hopf algebras
+
+In this file we define `HopfAlgebra`, and provide instances for:
+
+* Commutative semirings: `CommSemiring.toHopfAlgebra`
+
+# Main definitions
+
+* `HopfAlgebra R A` : the Hopf algebra structure on an `R`-bialgebra `A`.
+* `HopfAlgebra.antipode R A` : The `R`-linear map `A →ₗ[R] A`
+
+## References
+
+* <https://en.wikipedia.org/wiki/Hopf_algebra>
+
+-/
+
+suppress_compilation
+
+universe u v
+
+/-- A Hopf algebra over a commutative (semi)ring `R` is a bialgebra over `R` equipped with a linear
+map `S` (the antipode of the Hopf algebra) satisfying the antipode axioms. -/
+class HopfAlgebra (R : Type u) (A : Type v) [CommSemiring R] [Semiring A] extends
+    Bialgebra R A where
+  /-- The antipode of the Hopf algebra -/
+  S : A →ₗ[R] A
+  /-- The antipode axioms for a Hopf algebra -/
+  mSid : (LinearMap.mul' R A) ∘ₗ (S.rTensor A) ∘ₗ comul = (Algebra.linearMap R A) ∘ₗ counit
+  midS : (LinearMap.mul' R A) ∘ₗ (S.lTensor A) ∘ₗ comul = (Algebra.linearMap R A) ∘ₗ counit
```

Comment:
I don't understand the choice of these field names. Could you replace them with something with underscores?

Perhaps `mul_rTensor_comul`, etc?